presto-hive/src/main/java/com/facebook/presto/hive/RowIDCoercer.java

```diff
@@ -16,29 +16,39 @@
 import com.facebook.presto.common.Subfield;
 import com.facebook.presto.common.block.Block;
 import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.block.LazyBlock;
 import com.facebook.presto.common.predicate.TupleDomainFilter;
-import com.facebook.presto.common.type.BigintType;
 import com.facebook.presto.common.type.Type;
-import com.facebook.presto.common.type.VarbinaryType;
+import io.airlift.slice.Slice;
 import io.airlift.slice.Slices;
 
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
-import java.nio.charset.StandardCharsets;
 
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
 import static com.google.common.base.Preconditions.checkArgument;
+import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.util.Objects.requireNonNull;
 
 public class RowIDCoercer
         implements HiveCoercer
 {
-    private final byte[] rowIDPartitionComponent;
-    private final byte[] rowGroupID; // file name
+    private final ByteBuffer byteBuffer;
```

Comment:
nit: would be good to give these more descriptive names for what the byte buffer and slice represent